### PR TITLE
Methods

### PR DESCRIPTION
**Pull request by Juakata**

### CODE REVIEW

File: methods.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    result = true
+    my_each { |e| result = false if yield(e) }
+    result
+  end
+
+  def my_count
+    result = 0
+    my_each { |e| result += 1 if yield(e) }
+    result
+  end
```
Comment (maelfosso) [inline methods.rb:68]:
```ruby
irb(main):003:0> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
irb(main):004:0> ary.count
=> 4
irb(main):005:0> ary.my_count
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):5
        3: from methods.rb:48:in `my_count'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:48:in `block in my_count'
LocalJumpError (no block given (yield))
irb(main):006:0> ary.count(2)
=> 2
irb(main):007:0> ary.my_count(2)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):7
        1: from methods.rb:46:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):008:0> ary.count{ |x| x%2==0 }
=> 3
irb(main):009:0> ary.my_count{ |x| x%2==0 }
=> 3
```

File: methods.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    arr = []
+    my_each { |e| arr.push(e) if block.call(e) }
+    arr
+  end
+
+  def my_all?
+    result = true
+    my_each { |e| result = false unless yield(e) }
+    result
+  end
```
Comment (maelfosso) [inline methods.rb:36]:
```ruby
irb(main):012:0> %w[ant bear cat].all? { |word| word.length >= 4 }
=> false
irb(main):013:0> %w[ant bear cat].my_all? { |word| word.length >= 4 }
=> false
irb(main):014:0> [nil, true, 99].all? 
=> false
irb(main):015:0> [nil, true, 99].my_all? 
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):15
        3: from methods.rb:30:in `my_all?'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:30:in `block in my_all?'
LocalJumpError (no block given (yield))
```

File: methods.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  def my_any?
+    result = false
+    my_each { |e| result = true if yield(e) }
+    result
+  end
```
Comment (maelfosso) [inline methods.rb:46]:
```ruby
irb(main):020:0> %w[ant bear cat].any? { |word| word.length >= 4 }
=> true
irb(main):021:0> %w[ant bear cat].my_any? { |word| word.length >= 4 }
=> true
irb(main):022:0> [nil, true, 99].any?
=> true
irb(main):023:0> [nil, true, 99].my_any?
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):23
        3: from methods.rb:36:in `my_any?'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:36:in `block in my_any?'
LocalJumpError (no block given (yield))
```

File: methods.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  def my_none?
+    result = true
+    my_each { |e| result = false if yield(e) }
+    result
+  end
+
```
Comment (maelfosso) [inline methods.rb:57]:
```ruby
irb(main):034:0> [nil].none? 
=> true
irb(main):035:0> [nil].my_none? 
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):35
        3: from methods.rb:42:in `my_none?'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:42:in `block in my_none?'
LocalJumpError (no block given (yield))
irb(main):036:0> [nil, false].none? 
=> true
irb(main):037:0> [nil, false].my_none? 
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):37
        3: from methods.rb:42:in `my_none?'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:42:in `block in my_none?'
LocalJumpError (no block given (yield))
irb(main):038:0> [nil, false, true].none?
=> false
irb(main):039:0> [nil, false, true].my_none?
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):39
        3: from methods.rb:42:in `my_none?'
        2: from methods.rb:8:in `my_each'
        1: from methods.rb:42:in `block in my_none?'
LocalJumpError (no block given (yield))
```

File: methods.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    else
+      i = 1
+      result = self[0]
+      while i < length
+        result = block.call(result, self[i])
+        i += 1
+      end
+    end
+    result
+  end
```
Comment (maelfosso) [inline methods.rb:100]:
```ruby
irb(main):044:0> [5,6,7,8,9,10].inject { |sum, n| sum + n } 
=> 45
irb(main):045:0> [5,6,7,8,9,10].my_inject { |sum, n| sum + n } 
=> 45
irb(main):046:0> [5,6,7,8,9,10].inject(5) { |sum, n| sum + n } 
=> 50
irb(main):047:0> [5,6,7,8,9,10].my_inject(5) { |sum, n| sum + n } 
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):47
        1: from methods.rb:62:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):048:0> [5,6,7,8,9,10].inject(1) { |product, n| product * n }
=> 151200
irb(main):049:0> [5,6,7,8,9,10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):49
        1: from methods.rb:62:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

It was a very very good pleasure to read your code. Well done.
But before moving forward, there are some changes to do.

Please check the comment below, correct to changes suggested and kindly re-submit a new code review requests.
Thanks

**Review by bellom — APPROVED** [review]
Hi @Juakata,
Great work making all the necessary corrections from the previous review.
#### Congratulation :+1: 
_Goodluck in the next project._

**Pull request merged**